ibis/backends/base/__init__.py

```diff
@@ -369,6 +369,100 @@ def read_csv(
             f"{self.name} does not support direct registration of CSV data."
         )
 
+    @util.experimental
+    def to_parquet(
+        self,
+        expr: ir.Table,
+        path: str | Path,
+        *,
+        params: Mapping[ir.Scalar, Any] | None = None,
+        limit: int | str | None = None,
+        **kwargs: Any,
+    ) -> None:
+        """Write the results of executing the given expression to a parquet file.
+
+        This method is eager and will execute the associated expression
+        immediately.
+
+        Parameters
+        ----------
+        expr
+            The ibis expression to execute and persist to parquet.
+        path
+            The data source. A string or Path to the parquet file.
+        params
+            Mapping of scalar parameter expressions to value.
+        limit
+            An integer to effect a specific row limit. A value of `None` means
+            "no limit". The default is in `ibis/config.py`.
+        **kwargs
+            Additional keyword arguments passed to pyarrow.parquet.ParquetWriter
+
+        https://arrow.apache.org/docs/python/generated/pyarrow.parquet.ParquetWriter.html
+        """
+        _ = self._import_pyarrow()
+        import pyarrow.parquet as pq
```

Comment:
Just a nit, but we could have a `pq = self._import_parquet()` similarly to `_import_pyarrow()` if we want even better error reporting.

Reply:
What's the likelihood of a user having `pyarrow` installed without `parquet` / `csv` support built?  I know it's possible to build `pyarrow` without that, but are there any released binaries / wheels / whatever that don't have it built in?